src/classes/RD2_ScheduleService.cls

```diff
@@ -0,0 +1,120 @@
+/*
+    Copyright (c) 2019 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Recurring Donations
+* @description Class with methods to create, modify and visualize RecurringDonationSchedule__c records.
+*
+*/
+
+public with sharing class RD2_ScheduleService {
+
+    private npe03__Recurring_Donation__c parentRecurringDonation;
+    @TestVisible
+    private List<RecurringDonationSchedule__c> recurringDonationSchedules = new List<RecurringDonationSchedule__c>();
+
+    private void init(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> rdSchedules) {
+        parentRecurringDonation = rd;
+        if (rdSchedules != null) {
+            recurringDonationSchedules = rdSchedules;
+        }
+    }
+
+    public RD2_ScheduleService(npe03__Recurring_Donation__c rd) {
```

Comment:
Rather than overloading constructors, I'm a fan of the builder pattern. For example:
```
public RD2_ScheduleService() { }

public withRecurringDonation(npe03__Recurring_Donation__c rd) {
    parentRecurringDonation = rd;
}
public withSchedules(List<RecurringDonationSchedule__c> rdSchedules) {
    recurringDonationSchedules = rdSchedules;
}
```